convert2rhel/unit_tests/pkghandler_test.py

```diff
@@ -79,6 +71,58 @@ def __call__(self, cmd, print_cmd=True, print_output=True):
             self.called += 1
             return self.output, self.ret_code
 
+    class IsFileMocked(unit_tests.MockFunction):
+        def __init__(self, is_file):
+            self.is_file = is_file
+
+        def __call__(self, *args, **kwargs):
+            return self.is_file
+
+    class DumbCallableObject(unit_tests.MockFunction):
+        def __call__(self, *args, **kwargs):
+            return
+
+    class AskDumbCallableObject(unit_tests.MockFunction):
+        def __call__(self, *args, **kwargs):
+            sys.exit(1)
+
+    class GetSizeMocked(unit_tests.MockFunction):
+        def __init__(self, file_size):
+            self.file_size = file_size
+
+        def __call__(self, *args, **kwargs):
+            return self.file_size
+
+
+    @unit_tests.mock(os.path, "isfile", IsFileMocked(is_file=False))
+    def test_clear_yum_versionlock_plugin_not_enabled(self):
+        self.assertFalse(pkghandler.clear_yum_versionlock())
+
+    @unit_tests.mock(utils, "ask_to_continue", DumbCallableObject())
+    @unit_tests.mock(os.path, "isfile", IsFileMocked(is_file=True))
+    @unit_tests.mock(os.path, "getsize", GetSizeMocked(file_size=1))
+    @unit_tests.mock(pkghandler, "call_yum_cmd", CallYumCmdMocked())
+    @unit_tests.mock(utils.RestorableFile, "backup", DumbCallableObject)
+    @unit_tests.mock(utils.RestorableFile, "restore", DumbCallableObject)
+    def test_clear_yum_versionlock_user_says_yes(self):
+        pkghandler.clear_yum_versionlock()
+        self.assertEqual(pkghandler.call_yum_cmd.called, 1)
+
+    @unit_tests.mock(utils, "ask_to_continue", AskDumbCallableObject())
+    @unit_tests.mock(os.path, "isfile", IsFileMocked(is_file=True))
+    @unit_tests.mock(pkghandler, "call_yum_cmd", CallYumCmdMocked())
+    def test_clear_yum_versionlock_user_says_no(self):
+        self.assertRaises(SystemExit, pkghandler.clear_yum_versionlock)
+
```

Comment:
Could you please elaborate more?

Reply:
You need to assert that `call_yum_cmd()` was never called here.

Having multiple assertions avoids most false positives and ensures that the application runs as you expect it to.

For instance, ask yourself: would the tests pass if `call_yum_cmd()` was called before `ask_to_continue()`? If it would, then the tests are incorrect